core/trie/key.go

```diff
@@ -72,9 +74,11 @@ func (k *Key) Equal(other *Key) bool {
 	} else if k == nil || other == nil {
 		return false
 	}
+	// I think we can compare k.bitset directly to other.bitset without creating slice and call to bytes.Equal, wdyt?
 	return k.len == other.len && bytes.Equal(k.bitset[:], other.bitset[:])
 }
 
+// could you please explain what is going on here?
```

Comment:
it tests if a bit at a certain position is set or not.
Assume 0b101 as the key, you Test(0) would return true, Test(1) would return false